src/constants/chains.ts

```diff
@@ -0,0 +1,19 @@
+import { arbitrumNovaChainId, gnosisChainId } from '@shapeshiftoss/caip'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import { selectFeatureFlag } from 'state/slices/selectors'
+import { store } from 'state/store'
+// returns known ChainIds as an array, excluding the ones that are currently flagged off
+export const knownChainIds = Object.values(KnownChainIds).filter(chainId => {
+  const isArbitrumEnabled = selectFeatureFlag(store.getState(), 'Arbitrum')
+  const isArbitrumNovaEnabled = selectFeatureFlag(store.getState(), 'ArbitrumNova')
+  const isGnosisEnabled = selectFeatureFlag(store.getState(), 'Gnosis')
+  const isPolygonEnabled = selectFeatureFlag(store.getState(), 'Polygon')
+  const isOptimismEnabled = selectFeatureFlag(store.getState(), 'Optimism')
+  if (chainId === arbitrumNovaChainId && !isArbitrumNovaEnabled) return false
```

Comment:
reason not use KnownChainIds enum for all of the checks?

Reply:
Absolutely not, only me thiking of going with a programmatic enum and extending it initially before realizing that wouldn't work because enums are Record<number, any> under the hood and the index matters!
f131df4afb